projects/com.oracle.truffle.llvm.parser.impl/src/com/oracle/truffle/llvm/parser/impl/LLVMVisitor.java

```diff
@@ -372,28 +411,12 @@ private void allocateGlobals(List<EObject> objects) {
                 globalVarNodes.add(globalVarWrite);
             }
             if (globalVar.getName().equals("@llvm.global_ctors")) {
-                ResolvedArrayType type = (ResolvedArrayType) typeResolver.resolve(globalVar.getType());
-                int size = type.getSize();
-                Object allocGlobalVariable = findOrAllocateGlobal(globalVar);
-                ResolvedType structType = type.getContainedType(0);
-                int structSize = LLVMTypeHelper.getByteSize(structType);
-                for (int i = 0; i < size; i++) {
-                    LLVMExpressionNode globalVarAddress = factoryFacade.createLiteral(allocGlobalVariable, LLVMBaseType.ADDRESS);
-                    LLVMExpressionNode iNode = factoryFacade.createLiteral(i, LLVMBaseType.I32);
-                    LLVMExpressionNode structPointer = factoryFacade.createGetElementPtr(LLVMBaseType.I32, globalVarAddress, iNode, structSize);
-                    LLVMExpressionNode loadedStruct = factoryFacade.createLoad(structType, structPointer);
-                    ResolvedType functionType = structType.getContainedType(1);
-                    int indexedTypeLength = LLVMTypeHelper.getAlignmentByte(functionType);
-                    LLVMExpressionNode oneLiteralNode = factoryFacade.createLiteral(1, LLVMBaseType.I32);
-                    LLVMExpressionNode functionLoadTarget = factoryFacade.createGetElementPtr(LLVMBaseType.I32, loadedStruct, oneLiteralNode, indexedTypeLength);
-                    LLVMExpressionNode loadedFunction = factoryFacade.createLoad(functionType, functionLoadTarget);
-                    LLVMExpressionNode[] argNodes = new LLVMExpressionNode[]{factoryFacade.createFrameRead(LLVMBaseType.ADDRESS, getStackPointerSlot())};
-                    assert argNodes.length == factoryFacade.getArgStartIndex().get();
-                    LLVMNode functionCall = factoryFacade.createFunctionCall(loadedFunction, argNodes, LLVMBaseType.VOID);
-                    globalVarNodes.add(functionCall);
-                }
-            } else if (globalVar.getName().equals("llvm.global_dtors")) {
-                throw new AssertionError("destructors not yet supported!");
+                addFunctionAttribute(globalVar, globalVarNodes);
+            } else if (globalVar.getName().equals("@llvm.global_dtors")) {
+                // destructor functions need to be treated differently than other global
```

Comment:
I think we can remove this comment since it would not be clear how destructor functions and global variables could be related.